spec/statistics/statistical_test/chi_squared_test_spec.rb

```diff
@@ -82,4 +83,40 @@
       expect(result[:alternative]).to be false
     end
   end
+
+  describe '.calculate_expected_matrix' do
+    
+    it 'calculate expected values for a 2*3 contingency table of observed values' do
+      
+      observed = Matrix[[388,51692],[119,45633],[271,40040]]
+      result = described_class.calculate_expected_matrix(observed)
+
+      expect(result).to eq(Matrix[[(40518240/138143), (7153969200/138143)], [(35595056/138143), (6284723480/138143)], [(31361958/138143), (5537320515/138143)]])
+
+    end
+
+  end
+
+  describe '.test_of_independence' do
+
+    it 'calculate test of independence for a 2*3 contingency table' do
+
+      observed = Matrix[[388,51692],[119,45633],[271,40040]]
+      alpha = 0.05
+
+      expect do
+        result = described_class.test_of_independence(alpha, observed)
```

Comment:
there's a scope issue here, as `result` stops existing when `expect do/end` finish execution. I think it makes more sense to extract it to another `it` statement. Something like this would work:

```ruby
it 'does not raise an error when trying to do a calculation with matrices' do
  expect do
    described_class.test_of_independence(alpha, observed)
  end.not_to raise_error
end
```